js/todolistview.js

```diff
@@ -0,0 +1,394 @@
+        </div>
+        <input class="edit" value="${contents}"></input>
+    `;
+  // 추후 접근 편의를 위해 <li> 태그의 id를 해당 할 일의 고유값으로 설정.
+  li.id = _id;
+  li.innerHTML = newTodoHTMLElement;
+  // 해당 할 일의 순위값 설정, 이벤트 처리기 등록.
+  li.querySelector(
+    `option[value="${priority == "FIRST" ? 1 : priority == "SECOND" ? 2 : 0}"]`
+  ).toggleAttribute("selected");
+  li.querySelector("select").addEventListener(
+    "change",
+    changeTodoElementPriority
+  );
+  // 해당 할 일에 대한 클릭(완료 상태 토글, 삭제), 더블클릭(편집모드 진입), 키 입력(편집모드 종료, 변경내역 반영) 이벤트 처리기 등록.
+  li.addEventListener("click", onTodoElementClicked);
+  li.addEventListener("dblclick", onTodoElementDblclicked);
+  li.addEventListener("keyup", onTodoElementKeyupped);
+  todoList.append(li);
+  // 만약 해당 할 일이 완료된 할 일이라면 토글.
+  // 이번에는 Event를 발생시키지 않음!! 이벤트 처리기 때문에 토글이 풀리는 문제가 발생.
+  if (isCompleted) {
+    li.querySelector("input.toggle").setAttribute("checked", "");
+    li.classList.add("completed");
+  }
+  updateCountText();
+}
+
+// 모든 할 일들 제거.
+async function onRemoveAllTodoClick(event) {
+  if (await deteleAllTodoElement(getActiveUserID())) {
+    clearTodoList();
+  } else {
+    // notify user that operation is failed.
+    alert("할 일을 제거하는 데 실패했습니다!");
+  }
+}
+
+// 할 일을 클릭했을때 이벤트 위임.
+function onTodoElementClicked(event) {
+  if (!event.target) {
+    return;
+  }
+
+  if (
+    event.target.nodeName === "INPUT" &&
+    event.target.classList.contains("toggle")
+  ) {
+    toggleTodoElementStatus(event);
+  } else if (event.target.nodeName === "BUTTON") {
+    removeCurrentTodoElement(event);
+  }
+}
+
+// 할 일을 더블클릭했을때 이벤트 위임.
+function onTodoElementDblclicked(event) {
+  if (event.target && event.target.nodeName === "LABEL") {
+    toggleTodoElementMode(event);
+  }
+}
+
+// 할 일을 조작 중 키를 입력했을때 이벤트 위임.
+function onTodoElementKeyupped(event) {
+  if (event.target && event.target.nodeName === "INPUT") {
+    updateTodoEdit(event);
+  }
+}
+
+// 할 일의 우선순위 변경.
+async function changeTodoElementPriority({ target }) {
+  const selectedPriority = parseInt(target.value); // 선택된 option 값은 select 요소의 value로 존재!
+  const userID = getActiveUserID();
+  const todoElement = target.closest("li");
+  const todoElementID = todoElement.id;
+
+  const animationToggler = getTodoElementLoadingAnimationToggler(todoElement);
+  animationToggler();
+
+  const updatedTodoElement = await updateTodoElementPriority(
+    userID,
+    todoElementID,
+    selectedPriority == 0 ? "NONE" : selectedPriority == 1 ? "FIRST" : "SECOND"
+  );
+  // 서버측 결과에 따라 스타일, 선택 항목 적용.
+  switch (updatedTodoElement.priority) {
+    case "FIRST":
+      target.value = 1;
+      target.classList.remove("secondary");
+      target.classList.add("primary");
+      break;
+    case "SECOND":
+      target.value = 2;
+      target.classList.remove("primary");
+      target.classList.add("secondary");
+      break;
+    case "NONE":
+      target.value = 0;
+      target.classList.remove("primary");
+      target.classList.remove("secondary");
+  }
```

Comment:
classList 에서 remove 메서드를 수행할 때, 없는 클래스를 지우더라도 에러가 발생하지 않는다고 합니다. 
또한 하나의 remove 메서드에 여러 클래스명을 넣어서 동시에 지우는 것도 가능합니다!
primary 와 secondary 를 모두 삭제한 뒤 케이스에 맞게 필요한 클래스만 추가하는 방식으로 작성하시면 더 코드가 간결해질 것 같습니다! 

```suggestion
  target.classList.remove("primary", "secondary");
  switch (updatedTodoElement.priority) {
    case "FIRST":
      target.value = 1;
      target.classList.add("primary");
      break;
    case "SECOND":
      target.value = 2;
      target.classList.add("secondary");
      break;
    case "NONE":
      target.value = 0;
  }
```

Reply:
여러개를 지울 수 있는지는 몰랐네요. 훨씬 더 좋은 방법인것 같습니다. 감사합니다!